### PR TITLE
[Doc] Add descriptions for C++ Client API doc (ClientConfiguration.h + ConsumerConfiguration.h + ReaderConfiguration.h)

### PR DESCRIPTION
**Pull request by Anonymitaet**
Fix a part of https://github.com/apache/pulsar/issues/9282

Below are previews on my local environment:

- `ClientConfiguration.h`
image

- `ConsumerConfiguration.h`
image

- `ReaderConfiguration.h`
image

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@BewareMyPower @Huanli-Meng @Jennifer88huang could you please help review? Thanks

File: pulsar-client-cpp/include/pulsar/ClientConfiguration.h [hunk 1]
```diff
@@ -142,29 +142,84 @@ class PULSAR_PUBLIC ClientConfiguration {
...
     ClientConfiguration& setTlsAllowInsecureConnection(bool allowInsecure);
+
+    /**
+     * @return whether the Pulsar client accepts untrusted TLS certificates from brokers
+     */
     bool isTlsAllowInsecureConnection() const;
 
+    /**
+     * Configure whether it allows validating hostname verification when a client connects to a broker over
+     * TLS. <p> It validates the incoming x509 certificate and matches the provided hostname (CN/SAN) with the
```
Comment (BewareMyPower) [inline pulsar-client-cpp/include/pulsar/ClientConfiguration.h:187]: Please use an empty line to separate the short description and the detail description

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: pulsar-client-cpp/include/pulsar/ConsumerConfiguration.h [hunk 2]
```diff
@@ -253,9 +333,23 @@ class PULSAR_PUBLIC ConsumerConfiguration {
...
+    /**
+     * @return the time duration for the PatternMultiTopicsConsumer performs a pattern auto discovery
+     */
     int getPatternAutoDiscoveryPeriod() const;
 
+    /**
+     * @param subscriptionInitialPosition the initial position at which to set
+     * the cursor when subscribing to the topic for the first time.
+     *
+     * The default value is `InitialPositionLatest`.
```
Comment (BewareMyPower) [inline pulsar-client-cpp/include/pulsar/ConsumerConfiguration.h:346]: Please follow the doc format

```
/**
 * Short description
 *
 * (optional) detail description
 * @param ...
 */
```

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: pulsar-client-cpp/include/pulsar/ReaderConfiguration.h [hunk 3]
```diff
@@ -128,8 +182,27 @@ class PULSAR_PUBLIC ReaderConfiguration {
...
     long getUnAckedMessagesTimeoutMs() const;
 
+    /**
+     * Set the tick duration time that defines the granularity of the ack-timeout redelivery (in
+     * milliseconds).
+     *
+     * <p>
+     *
+     * The default value is 1000, which means 1 second.
+     *
```
Comment (BewareMyPower) [inline pulsar-client-cpp/include/pulsar/ReaderConfiguration.h:192]: These empty lines could be removed.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

**Comment by BewareMyPower** [comment]
I've left a few comments, PTAL.

**Comment by Anonymitaet** [comment]
@BewareMyPower many thanks for your guidance! I've incorporated ur comments, PTAL.

**Comment by BewareMyPower** [comment]
Please format your file before pushing your commits.

**Review by BewareMyPower — APPROVED** [review]

**Review by Jennifer88huang — APPROVED** [review]

**Pull request merged**